src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs

```diff
@@ -181,16 +183,21 @@ private ICollection<FileInfo> PutFilesToRightModuleOrRoot(IEnumerable<ProjectDat
                 if (!file.Exists)
                 {
                     logger.LogWarning(Resources.WARN_FileDoesNotExist, file);
-                    logger.LogDebug(Resources.DEBUG_FileReferencedByProjects, string.Join("', '",
-                        group.Value.Select(x => x.Project.FullPath)));
+                    logger.LogDebug(
+                        Resources.DEBUG_FileReferencedByProjects,
+                        string.Join("', '", group.Value.Select(x => x.Project.FullPath)));
                     continue;
                 }
 
                 if (!PathHelper.IsInDirectory(file, baseDirectory)) // File is outside of the SonarQube root module
                 {
-                    logger.LogWarning(Resources.WARN_FileIsOutsideProjectDirectory, file, baseDirectory.FullName);
-                    logger.LogDebug(Resources.DEBUG_FileReferencedByProjects, string.Join("', '",
-                        group.Value.Select(x => x.Project.FullPath)));
+                    if (!file.FullName.Contains(Path.Combine(".nuget", "packages")))
```

Comment:
can you please add tests with a `.CS` file outside the base dir which are inside:
- a `.nuget` folder 
- a `packages` folder
- a `.nugetpackages` folder (no separator)
- a `.nuget/foo/packages` folder 

all these should generate warnings

Also, can you add a test with a file inside `.nuget/cache` inside the base dir (should not raise warning)?

What about `.NUGET/CACHE`? should the comparison be case insensitive?

Reply:
The default NuGet cache path is `.nuget/cache` (in both Unix and Windows). 
Of course, the user can create a custom cache that can be with caps or with a totally different name.
IMO only the default should be supported. 
Wdyt?

Reply:
Ok, makes sense RE: case sensitivity.
Still good to have a test ;)